opensrp-app/src/main/java/org/smartregister/repository/TaskRepository.java

```diff
@@ -112,6 +121,18 @@ public static void createTable(SQLiteDatabase database) {
         database.execSQL(CREATE_TASK_PLAN_GROUP_INDEX);
     }
 
+    /**
+     * Migrate the older database by add restriction columns and changing of priority to enum
+     *
+     * @param database the database being upgraded
+     */
+    public static void updatePriorityToEnumAndAddRestrictions(@NonNull SQLiteDatabase database) {
+        DatabaseMigrationUtils.addColumnIfNotExists(database, TASK_TABLE, RESTRICTION_REPEAT, VARCHAR);
+        DatabaseMigrationUtils.addColumnIfNotExists(database, TASK_TABLE, RESTRICTION_START, VARCHAR);
+        DatabaseMigrationUtils.addColumnIfNotExists(database, TASK_TABLE, RESTRICTION_END, VARCHAR);
+        DatabaseMigrationUtils.recreateSyncTableWithExistingColumnsOnly(database, TASK_TABLE, COLUMNS, CREATE_TASK_TABLE);
+        database.execSQL(String.format("UPDATE %s SET %s=?", TASK_TABLE, PRIORITY), new Object[]{Task.TaskPriority.ROUTINE.name()});
```

Comment:
on line 92 Priority has been changed from an integer field to a varchar. Do we need a migration for that?

Reply:
The table is being recreated using `DatabaseMigrationUtils.recreateSyncTableWithExistingColumnsOnly` above, this is because sqlite does not support alter column data types or dropping columns, the table must be recreated